pkg/deployment/os.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"context"
+
+	dataplaneutil "github.com/openstack-k8s-operators/dataplane-operator/pkg/util"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	ansibleeev1alpha1 "github.com/openstack-k8s-operators/openstack-ansibleee-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// InstallOS ensures the node Operating System is installed
+func InstallOS(ctx context.Context, helper *helper.Helper, obj client.Object, sshKeySecret string, inventoryConfigMap string) error {
+
+	tasks := []dataplaneutil.Task{
+		{
+			Name:          "Install edpm_bootstrap",
+			RoleName:      "edpm_bootstrap",
+			RoleTasksFrom: "bootstrap.yml",
+			Tags:          []string{"edpm_bootstrap"},
+		},
+		{
+			Name:     "Install edpm_kernel",
+			RoleName: "edpm_kernel",
```

Comment:
missing a RoleTasksFrom here which causes a traceback:

`1.6770903274761724e+09  ERROR   Unable to execute Ansible for InstallOS {"controller": "openstackdataplanenode", "controllerGroup": "dataplane.openstack.org", "controllerKin
d": "OpenStackDataPlaneNode", "OpenStackDataPlaneNode": {"name":"openstackdataplanenode-sample-deployment","namespace":"openstack"}, "namespace": "openstack", "name": "opens
tackdataplanenode-sample-deployment", "reconcileID": "1d465598-2b43-4528-b357-5eca6004f550", "error": "OpenStackAnsibleEE.ansibleee.openstack.org \"dataplane-deployment-inst
all-osgg26f\" is invalid: spec.roles.tasks[1].import_role.tasks_from: Required value"}                                                                                       
github.com/openstack-k8s-operators/dataplane-operator/pkg/deployment.InstallOS                                                                                               
        /home/jslagle/code/github/openstack-k8s-operators/dataplane-operator/pkg/deployment/os.go:80                                                                         
github.com/openstack-k8s-operators/dataplane-operator/pkg/deployment.ConditionalDeploy                                                                                       
        /home/jslagle/code/github/openstack-k8s-operators/dataplane-operator/pkg/deployment/deployment.go:119                                                                
github.com/openstack-k8s-operators/dataplane-operator/pkg/deployment.Deploy                                                                                                  
`